lib/iex/lib/iex/evaluator.ex

```diff
@@ -24,6 +24,21 @@ defmodule IEx.Evaluator do
     end
   end
 
+  @doc """
+  Gets a value out of the binding, using the provided
+  variable name and map key path.
+  """
+  @spec value_from_binding(pid, atom, [atom]) :: {:ok, any} | :error
+  def value_from_binding(evaluator, var_name, map_key_path) do
+    send evaluator, {:value_from_binding, self(), var_name, map_key_path}
```

Comment:
The good practice here is to build a reference, send it to the server, and match on it on the response:

```elixir
ref = make_ref()
send evaluator, {:value_from_binding, ref, self(), var_name, map_key_path}

receive do
  {^ref, result} -> result
after
  5000 -> :error
end
```

This is for many reasons:

  1. If for some reason the evaluator cannot reply after 5000 miliseconds, it will eventually reply and a future request for a binding may return the value of a previous request (which will be sitting in the autocomplete process inbox)

  2. the VM can optimize that code snippet really well. It knows the time the reference was built and because they are unique, it inserts markers in the current process message queue to avoid traversing it from the beginning when looking back for the reply